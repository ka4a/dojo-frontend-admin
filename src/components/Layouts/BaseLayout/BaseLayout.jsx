import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Toast, Icon, Modal } from '@edx/paragon';
import { useDispatch, useSelector } from 'react-redux';
import { closeToast } from '../../../store/baseLayout/baseLayout.slice';
import { getToastContent } from '../../../store/baseLayout/baseLayout.selector';
import { requestStatus } from '../../../utils/constants/requestStatusConstants';
import './BaseLayout.styles.scss';

const BaseLayout = ({
  children, statuses, errors, position,
}) => {
  const dispatch = useDispatch();
  const toast = useSelector(getToastContent);

  return (
    <div className="base-layout">
      {statuses.some((status) => status === requestStatus.LOADING) && (
        <h2>Loading...</h2>
      )}
      {errors.some((error) => error) && (
        <Modal
          open
          title="An error occurred:"
          body={errors?.find((error) => error)}
          onClose={() => {}}
          closeText={<Icon className="fa fa-ship" screenReaderText="Close" />}
        />
      )}
      {statuses.every((status) => status === requestStatus.RESOLVED) && (
        <div className={classNames({ [position]: position })}>{children}</div>
      )}
      <Toast
        onClose={() => {
          dispatch(closeToast());
        }}
        show={toast.isOpened}
      >
        {toast.body && toast.body}
      </Toast>
    </div>
  );
};

BaseLayout.propTypes = {
  statuses: PropTypes.arrayOf(PropTypes.string),
  errors: PropTypes.arrayOf(PropTypes.string),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  position: PropTypes.string,
};

BaseLayout.defaultProps = {
  position: '',
  statuses: [],
  errors: [],
  children: null,
};

export default BaseLayout;
