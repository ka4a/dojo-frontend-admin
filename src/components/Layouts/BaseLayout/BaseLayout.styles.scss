.base-layout {
  .centered,
  .left,
  .right {
    max-width: 1280px;
  }

  .centered {
    margin: 0 auto;
  }

  .left {
    margin: auto 0 0 0;
  }

  .right {
    margin: 0 0 0 auto;
  }
}
