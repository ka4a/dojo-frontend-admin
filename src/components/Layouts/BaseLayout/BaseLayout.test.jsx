import { shallow, mount } from 'enzyme';
import { AdminDashboardProvider } from '../../../test/AdminDashboardProvider';
import BaseLayout from './BaseLayout';

const defaultProps = {
  statuses: [],
  errors: [],
  position: 'center',
};

describe('BaseLayout', () => {
  it('should render BaseLayout component', () => {
    shallow(
      <AdminDashboardProvider>
        <BaseLayout {...defaultProps}>Test</BaseLayout>
      </AdminDashboardProvider>,
    );
  });

  it('it should render the expected HTML', () => {
    expect(
      mount(
        <AdminDashboardProvider>
          <BaseLayout {...defaultProps}>Test</BaseLayout>
        </AdminDashboardProvider>,
      ).html(),
    ).toMatchSnapshot();
  });
});
