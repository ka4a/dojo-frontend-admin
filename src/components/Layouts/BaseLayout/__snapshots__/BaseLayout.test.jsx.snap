// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout it should render the expected HTML 1`] = `"<div class=\\"base-layout\\"><div class=\\"center\\">Test</div></div>"`;
